### ISSUE TITLE
Error in .subset2

### ISSUE THREAD

**Issue by hmeleiro** [jacob-long/interactions#11]
I'm having trouble using the ``` interact_plot()``` function. The following command used to work correctly when this function was part of ```jtools```package:

```
interact_plot(model2, pred = Africana, modx = tipo, interval = T, robust = "HC1")
```

But now I get this error:

```
Error in .subset2(x, i, exact = exact) : invalid subscript type 'symbol'
```

This is the model summary:

```
> summary(model2)

Call:
lm(formula = H2.inmigracion, data = p@data)

Residuals:
     Min       1Q   Median       3Q      Max 
-17.3044  -1.1706  -0.1362   0.9745  17.0160 

Coefficients:
                                                      Estimate  Std. Error t value           Pr(>|t|)    
(Intercept)                                         5.88028187  0.38103225  15.433            < 2e-16 ***
renta_disponible_media16                            0.00005651  0.00001435   3.939 0.0000827893452961 ***
tipoZona de densidad intermedia                    -0.74167836  0.19023050  -3.899 0.0000977235555040 ***
tipoCiudades                                       -1.51741257  0.26711938  -5.681 0.0000000140570307 ***
alquiler                                           -0.01124921  0.00352482  -3.191           0.001423 ** 
Africana                                            0.15533060  0.02790134   5.567 0.0000000270340785 ***
masde64                                            -0.05929514  0.00298283 -19.879            < 2e-16 ***
VOX15                                               0.76708174  0.05843900  13.126            < 2e-16 ***
derecha                                             0.13950679  0.00398449  35.012            < 2e-16 ***
izquierda                                          -0.07595612  0.00428095 -17.743            < 2e-16 ***
paro.hom                                           -0.14380445  0.02347032  -6.127 0.0000000009527316 ***
paro.hom.diff06.18                                  0.14928709  0.04180212   3.571           0.000358 ***
tipoZona de densidad intermedia:paro.hom.diff06.18  0.22277924  0.04830963   4.611 0.0000040820913076 ***
tipoCiudades:paro.hom.diff06.18                     0.48334043  0.06332275   7.633 0.0000000000000266 ***
tipoZona de densidad intermedia:Africana           -0.00419771  0.02891354  -0.145           0.884573    
tipoCiudades:Africana                              -0.17692442  0.03016461  -5.865 0.0000000047257359 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 1.999 on 5918 degrees of freedom
  (6 observations deleted due to missingness)
Multiple R-squared:  0.6372,	Adjusted R-squared:  0.6362 
F-statistic: 692.8 on 15 and 5918 DF,  p-value: < 2.2e-16
```

And this is my ``` sessionInfo()```:

```
> sessionInfo()
R version 3.4.1 (2017-06-30)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: OS X El Capitan 10.11.6

Matrix products: default
BLAS: /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.4/Resources/lib/libRlapack.dylib

locale:
[1] es_ES.UTF-8/es_ES.UTF-8/es_ES.UTF-8/C/es_ES.UTF-8/es_ES.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] jtools_2.0.0    readxl_1.2.0    spdep_0.8-1     spData_0.3.0    Matrix_1.2-15   raster_2.8-19   sp_1.3-1        forcats_0.3.0  
 [9] stringr_1.4.0   dplyr_0.8.0     purrr_0.3.0     readr_1.3.1     tidyr_0.8.2     tibble_2.0.1    ggplot2_3.1.0   tidyverse_1.2.1

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.0         lubridate_1.7.4    lattice_0.20-38    deldir_0.1-16      zoo_1.8-4          gtools_3.8.1       assertthat_0.2.0  
 [8] R6_2.4.0           cellranger_1.1.0   plyr_1.8.4         backports_1.1.3    stats4_3.4.1       coda_0.19-2        httr_1.4.0        
[15] pillar_1.3.1       rlang_0.3.1        lazyeval_0.2.1     rstudioapi_0.9.0   gdata_2.18.0       gmodels_2.18.1     splines_3.4.1     
[22] rgdal_1.3-6        munsell_0.5.0      broom_0.5.1        compiler_3.4.1     modelr_0.1.3       xfun_0.4           pkgconfig_2.0.2   
[29] tidyselect_0.2.5   expm_0.999-3       codetools_0.2-16   crayon_1.3.4       withr_2.1.2        MASS_7.3-51.1      grid_3.4.1        
[36] nlme_3.1-137       jsonlite_1.6       gtable_0.2.0       huxtable_4.3.0     magrittr_1.5       scales_1.0.0.9000  cli_1.0.1         
[43] stringi_1.3.1      LearnBayes_2.15.1  xml2_1.2.0         generics_0.0.2     boot_1.3-20        sandwich_2.5-0     tools_3.4.1       
[50] glue_1.3.0         hms_0.4.2          yaml_2.2.0         colorspace_1.4-0   rvest_0.3.2        knitr_1.21         haven_2.0.0       
[57] interactions_1.0.0
```

**Comment by jacob-long** [comment]
Thanks for this report! There's one more thing you can do that should help me figure this out. After you run the code that causes the error, run the `traceback()` function and paste the output here.

**Comment by hmeleiro** [comment]
Thanks for the quick response. Here is the traceback:

```
> interact_plot(model2, pred = Africana, modx = tipo, data = p@data, interval = T, robust = "HC1")
Error in .subset2(x, i, exact = exact) : invalid subscript type 'symbol'
> traceback()
8: (function(x, i, exact) if (is.matrix(i)) as.matrix(x)[[i]] else .subset2(x, 
       i, exact = exact))(x, ..., exact = exact)
7: `[[.data.frame`(d, pred)
6: d[[pred]]
5: unique(d[[pred]])
4: plot_mod_continuous(predictions = pm, pred = pred, modx = modx, 
       resp = resp, mod2 = mod2, data = d, plot.points = plot.points | 
           partial.residuals, interval = interval, linearity.check = linearity.check, 
       x.label = x.label, y.label = y.label, pred.labels = pred.labels, 
       modx.labels = modx.labels, mod2.labels = mod2.labels, main.title = main.title, 
       legend.main = legend.main, colors = colors, line.thickness = line.thickness, 
       vary.lty = vary.lty, jitter = jitter, modxvals2 = modxvals2, 
       mod2vals2 = mod2vals2, weights = weights, rug = rug, rug.sides = rug.sides, 
       point.size = point.size, point.shape = point.shape, facet.modx = facet.modx)
3: (function (model, pred, modx, modx.values = NULL, mod2 = NULL, 
       mod2.values = NULL, centered = "all", data = NULL, plot.points = FALSE, 
       interval = FALSE, int.type = c("confidence", "prediction"), 
       int.width = 0.95, outcome.scale = "response", linearity.check = FALSE, 
       facet.modx = FALSE, robust = FALSE, cluster = NULL, vcov = NULL, 
       set.offset = 1, x.label = NULL, y.label = NULL, pred.labels = NULL, 
       modx.labels = NULL, mod2.labels = NULL, main.title = NULL, 
       legend.main = NULL, colors = NULL, line.thickness = 1, vary.lty = TRUE, 
       point.size = 1.5, point.shape = FALSE, jitter = 0, rug = FALSE, 
       rug.sides = "b", partial.residuals = FALSE, point.alpha = 0.6, 
       color.class = NULL, ...) 
   {
       dots <- list(...)
       if (length(dots) > 0) {
           if ("modxvals" %in% names(dots)) {
               modx.values <- dots$modxvals
           }
           if ("mod2vals" %in% names(dots)) {
               mod2.values <- dots$mod2vals
           }
       }
       if (!is.null(color.class)) {
           colors <- color.class
           msg_wrap("The color.class argument is deprecated. Please use 'colors'\n             instead.")
       }
       pred <- quo_name(enexpr(pred))
       modx <- quo_name(enexpr(modx))
       if (modx == "NULL") {
           modx <- NULL
       }
       mod2 <- quo_name(enexpr(mod2))
       if (mod2 == "NULL") {
           mod2 <- NULL
       }
       if (any(c(pred, modx, mod2) %in% centered)) {
           warn_wrap("You cannot mean-center the focal predictor or moderators with\n              this function.")
           centered <- centered %not% c(pred, modx, mod2)
           if (length(centered) == 0) {
               centered <- "none"
           }
       }
       modxvals2 <- mod2vals2 <- resp <- NULL
       if (linearity.check == TRUE) {
           facet.modx <- TRUE
       }
       if (is.null(data)) {
           d <- get_data(model, warn = TRUE)
       }
       else {
           d <- data
       }
       weights <- get_weights(model, d)$weights_name
       pred_out <- prep_data(model = model, pred = pred, modx = modx, 
           modx.values = modx.values, mod2 = mod2, mod2.values = mod2.values, 
           centered = centered, interval = interval, int.type = int.type, 
           int.width = int.width, outcome.scale = outcome.scale, 
           linearity.check = linearity.check, robust = robust, cluster = cluster, 
           vcov = vcov, set.offset = set.offset, modx.labels = modx.labels, 
           mod2.labels = mod2.labels, facet.modx = facet.modx, d = d, 
           survey = "svyglm" %in% class(model), weights = weights, 
           preds.per.level = 100, partial.residuals = partial.residuals, 
           ...)
       meta <- attributes(pred_out)
       lapply(names(meta), function(x, env) {
           env[[x]] <- meta[[x]]
       }, env = environment())
       pm <- pred_out$predicted
       d <- pred_out$original
       if (!is.numeric(d[[pred]])) {
           stop("Focal predictor (\"pred\") cannot be a factor. Either", 
               " use it as modx, convert it to a numeric dummy variable,", 
               " or use the cat_plot function for factor by factor interaction", 
               " plots.")
       }
       plot_mod_continuous(predictions = pm, pred = pred, modx = modx, 
           resp = resp, mod2 = mod2, data = d, plot.points = plot.points | 
               partial.residuals, interval = interval, linearity.check = linearity.check, 
           x.label = x.label, y.label = y.label, pred.labels = pred.labels, 
           modx.labels = modx.labels, mod2.labels = mod2.labels, 
           main.title = main.title, legend.main = legend.main, colors = colors, 
           line.thickness = line.thickness, vary.lty = vary.lty, 
           jitter = jitter, modxvals2 = modxvals2, mod2vals2 = mod2vals2, 
           weights = weights, rug = rug, rug.sides = rug.sides, 
           point.size = point.size, point.shape = point.shape, facet.modx = facet.modx)
   })(model = model2, pred = Africana, modx = tipo, data = p@data, 
       interval = T, robust = "HC1")
2: do.call(get("interact_plot", asNamespace("interactions")), as.list(match.call(get("interact_plot", 
       asNamespace("interactions"))))[-1])
1: interact_plot(model2, pred = Africana, modx = tipo, data = p@data, 
       interval = T, robust = "HC1")
```


By the way, I realized that this error came up with a simple regression with the ``` mpg``` dataset so I guess the problem is in my R session. I'll try tomorrow in the computer I have at work.

```
> library(interactions)
> library(ggplot2)
> model <- lm(cty ~ hwy + drv + hwy*drv, data = mpg)
> interact_plot(model, pred = hwy, modx = drv, interval =  T, robust = "HC1")
Error in .subset2(x, i) : invalid subscript type 'symbol'
```

**Comment by jacob-long** [comment]
Have you been able to replicate this on another computer? I haven't had any luck.

**Comment by hmeleiro** [comment]
I just installed the package in another computer and it works. There may be something wrong with the other computer. Maybe you can just clse the issue and reopen if somebody else gets the same error.

Great package by the way.

**Comment by rachelmtheodore** [comment]
In case it's helpful... I had the same issue. I got this error:

```Error in .subset2(x, i, exact = exact) : invalid subscript type 'symbol'```

when running ```interact_plot()``` with the ```interactions``` package for code that used to work with ```jtools```. I had installed ```interactions``` today using ```devtools::install_github("jacob-long/interactions")```.

The issue was resolved by updating my R version to ```3.5.3```; I had been using R version ```3.4.0```.

I love your packages, by the way - thanks!

**Comment by tcdennehy** [comment]
I'm also getting this error in both interactions and jtools.

> Error in .subset2(x, i) : invalid subscript type 'symbol'

I updated R and R Studio as far as I could given my operating system (OS X Yosemite; latest compatible versions are R Studio 1.1.463 and R R 3.3.3). I also downloaded the latest versions of each package using devtools.

The issue has rendered both jtools and interactions virtually unusable. Any recommendations for people using older operating systems? [Upgrading the OS is a much larger project than I have time for given that I'll have to clear up room on my hard drive.]

**Comment by alecristia** [comment]
This code used to work, but then I updated my R to 3.6.3, and now I'm getting this error for both interact_plot and cat_plot. For me, it happens only if I declare weights.

> traceback()
> 6: (function(x, i, exact) if (is.matrix(i)) as.matrix(x)[[i]] else .subset2(x, 
       i, exact = exact))(x, ..., exact = exact)
> 5: `[[.data.frame`(data, wname)
> 4: data[[wname]]
> 3: get_weights(model, d)
> 2: get_data(model, warn = TRUE)
> 1: interact_plot(model = pred_in_ovl, modx = monoling, pred = age_s, 
       plot.points = TRUE, point.size = 0.5, point.alpha = 0.1)

**Comment by jhemberger** [comment]
I'm in the same boat as alecristia I was using the interact_plot successfully yesterday (models included weights), but I re-ran the models today and get the same error. 

`number of items to replace is not a multiple of replacement length the condition has length > 1 and only the first element will be usedError in .subset2(x, i, exact = exact) : subscript out of bounds`

I found out that it was due to a naming convention of the weighting variable. It didn't work when weight variable name contained a period (e.g., my.weights), but did if it was instead an underscore (e.g., my_weights).

Also, loving both the jtools and interactions packages. Bravo!
